healthScore/views.py

```diff
@@ -511,3 +515,108 @@ def hospital_staff_directory(request):
         "get_facility_admins_url": "api/get-facility-admins/",
     }
     return render(request, "healthcare_facility.html", context)
+
+
+@login_required
+@csrf_exempt
+def add_healthcare_staff(request):
+    if request.user.is_authenticated and request.method == "POST":
+        email = request.POST.get("email")
+        fullname = request.POST.get("fullname")
+        contactInfo = request.POST.get("contactInfo")
+        is_admin = int(request.POST.get("is_admin"))
+        specialization = request.POST.get("specialization")
+        hospital_id = request.POST.get("hospital_id")
+
+        context = {"error_message:": ""}
+
+        if User.objects.filter(email=email).exists():
+            user = User.objects.get(email=email)
+            if user.is_patient:
+                context["error_message"] = (
+                    "A patient account already exists with this email"
+                )
+            elif user.is_staff:
+                context["error_message"] = (
+                    "An admin account already exists with this email"
+                )
+            else:
+                context["error_message"] = (
+                    "A healthcare worker account already exists with this email"
+                )
+
+            return render(request, "healthcare_facility.html", context)
+
+        user_fields = {
+            "email": email,
+            "password": "dummy_password",  # healthcare worker will have to reset the password on first login
+            "name": fullname,
+            "contactInfo": contactInfo,
+        }
+
+        hospital = get_object_or_404(Hospital, id=hospital_id)
+
+        user = None
+        if is_admin:
+            user = User.objects.create_staff(**user_fields)
+        else:
+            user = User.objects.create_healthcare_worker(**user_fields)
+
+        HospitalStaff.objects.create(
+            hospitalID=hospital,
+            admin=is_admin,
+            name=fullname,
+            specialization=specialization,
+            contactInfo=contactInfo,
+            userID=user.id,
+        )
+
+        return redirect("hospital_staff_directory")
+
+    return JsonResponse({"error": "Unauthorized"}, status=401)
+
+
+@login_required
+@csrf_exempt
+def deactivate_healthcare_staff(request):
+    if request.user.is_authenticated and request.method == "PUT":
+        updatedData = json.loads(request.body)
+        user_id = updatedData.get("user_id")
+
+        user = get_object_or_404(User, id=user_id)
+
+        if user.is_patient:
+            return render(
+                request,
+                "healthcare_facility.html",
+                {"error_message": "Patient's account cannot be edited"},
+            )
+
+        user.is_active = 0
```

Comment:
is_active is a bool value. So could we set this to False instead of 0? It might cos data mismatch issues since the default values for this field is True

Reply:
Yes it is a bool value, but it is set in the DB as a tinyint (0 or 1). That's why I set it as 0
<img width="1096" alt="image" src="https://github.com/gcivil-nyu-org/INT2-Monday-Spring2024-Team-1/assets/83601608/db689f03-2339-4617-866c-8181df94e2b7">